projection/ibc_channel/view/ibc_channels.go

```diff
@@ -564,6 +565,167 @@ func (ibcChannelsView *IBCChannels) List(
+	if err != nil {
+		return nil, nil, fmt.Errorf("error building channels select SQL: %v, %w", err, rdb.ErrBuildSQLStmt)
+	}
+
+	rowsResult, err := ibcChannelsView.rdb.Query(sql, sqlArgs...)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error executing channels select SQL: %v: %w", err, rdb.ErrQuery)
+	}
+	defer rowsResult.Close()
+
+	// key: chainID
+	// value: []IBCChannelRow
+	chainChannelsMap := make(map[string][]IBCChannelRow)
+
+	for rowsResult.Next() {
+		var channel IBCChannelRow
+		var bondedTokensJSON string
+		lastActivityTimeReader := ibcChannelsView.rdb.NtotReader()
+		createdAtTimeReader := ibcChannelsView.rdb.NtotReader()
+		if err = rowsResult.Scan(
+			&channel.ChannelID,
+			&channel.PortID,
+			&channel.ConnectionID,
+			&channel.CounterpartyChannelID,
+			&channel.CounterpartyPortID,
+			&channel.CounterpartyChainID,
+			&channel.Status,
+			&channel.PacketOrdering,
+			&channel.LastInPacketSequence,
+			&channel.LastOutPacketSequence,
+			&channel.TotalTransferInCount,
+			&channel.TotalTransferOutCount,
+			&channel.TotalTransferOutSuccessCount,
+			&channel.TotalTransferOutSuccessRate,
+			createdAtTimeReader.ScannableArg(),
+			&channel.CreatedAtBlockHeight,
+			&channel.Verified,
+			&channel.Description,
+			lastActivityTimeReader.ScannableArg(),
+			&channel.LastActivityBlockHeight,
+			&bondedTokensJSON,
+		); err != nil {
+			if errors.Is(err, rdb.ErrNoRows) {
+				return nil, nil, rdb.ErrNoRows
+			}
+			return nil, nil, fmt.Errorf("error scanning channel row: %v: %w", err, rdb.ErrQuery)
+		}
+
+		createdAtBlockTime, parseErr := createdAtTimeReader.Parse()
+		if parseErr != nil {
+			return nil, nil, fmt.Errorf("error parsing createdAtBlockTime: %v: %w", parseErr, rdb.ErrQuery)
+		}
+		channel.CreatedAtBlockTime = *createdAtBlockTime
+
+		lastActivityBlockTime, parseErr := lastActivityTimeReader.Parse()
+		if parseErr != nil {
+			return nil, nil, fmt.Errorf("error parsing lastActivityBlockTime: %v: %w", parseErr, rdb.ErrQuery)
+		}
+		channel.LastActivityBlockTime = *lastActivityBlockTime
+
+		if err = jsoniter.UnmarshalFromString(bondedTokensJSON, &channel.BondedTokens); err != nil {
+			return nil, nil, fmt.Errorf("error unmarshalling channel bonded_tokens JSON: %v: %w", err, rdb.ErrQuery)
+		}
+
+		// Append channel to chainChannelsMap[chainID]
+		if channels, ok := chainChannelsMap[channel.CounterpartyChainID]; ok {
+			channels = append(channels, channel)
+			chainChannelsMap[channel.CounterpartyChainID] = channels
+		} else {
+			newChannels := make([]IBCChannelRow, 0)
+			newChannels = append(newChannels, channel)
+			chainChannelsMap[channel.CounterpartyChainID] = newChannels
+		}
+	}
+
+	// Convert map[string][]IBCChannelRow to []ChainChannels
+	var chainChannelsList []ChainChannels
+	for chainID, channels := range chainChannelsMap {
+		chainChannels := ChainChannels{
+			ChainID:  chainID,
+			Channels: channels,
+		}
+		chainChannelsList = append(chainChannelsList, chainChannels)
+	}
```

Comment:
Again this can be transform to a function

Reply:
Thanks for the advice, let me update the code 🙏